com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/assist/JsonReferenceProposalProvider.java

```diff
@@ -112,66 +125,103 @@ public String value() {
+            if (parameters == null) {
+                return results;
+            }
+
+            final String basePath = (path != null ? path.toString() : "") + "#/" + value() + "/";
+
+            for (Iterator<String> it = parameters.fieldNames(); it.hasNext();) {
+                String key = it.next();
+                String value = basePath + key.replaceAll("/", "~1");
+                String encoded = URLUtils.encodeURL(value);
+
+                results.add(new Proposal("\"" + encoded + "\"", key, null, value));
+            }
+
+            return results;
+        }
+
         public static ContextTypeCollection newContentTypeCollection(Iterable<ContextType> contextTypes) {
-        	return new ContextTypeCollection(contextTypes);
+            return new ContextTypeCollection(contextTypes);
+        }
+
+        public static ContextTypeCollection emptyContentTypeCollection() {
+            return new ContextTypeCollection(Collections.<ContextType> emptyList());
         }
-        
-		public static ContextTypeCollection emptyContentTypeCollection() {
-			return new ContextTypeCollection(Collections.<ContextType>emptyList());
-		}
-    }
-    
-    public static class ContextTypeCollection {
-    	
-    	private final Iterable<ContextType> contextTypes;
-
-		protected ContextTypeCollection(Iterable<ContextType> contextTypes) {
-			this.contextTypes = contextTypes;
-    	}
-		
-		public ContextType get(String path) {
-			if (Strings.emptyToNull(path) == null) {
-				return ContextType.UNKNOWN;
-			}
-			for (ContextType next : contextTypes) {
-				if (path.matches(next.regex)) {
-					return next;
-				}
-			}
-			return ContextType.UNKNOWN;
-		}
     }
 
-    /**
-     * Returns all proposals found in the document at the specified field.
-     * 
-     * @param document
-     * @param fieldName
-     * @param path
-     * @return Collection of proposals
-     */
-    public Collection<Proposal> collectProposals(JsonNode document, String fieldName, IPath path) {
-        final Collection<Proposal> results = Lists.newArrayList();
-        if (fieldName == null) {
-            return results;
+    public static class ContextTypeValue extends ContextType {
+
+        public ContextTypeValue(String value, String label, String regex) {
+            super(value, label, regex);
+        }
+
+        public ContextTypeValue(String value, String label, String regex, boolean isLocalOnly) {
+            super(value, label, regex, isLocalOnly);
         }
 
-		final JsonNode parameters = ValidationUtil.findNode(fieldName, document);
-        if (parameters == null) {
-        	return results;
+        @Override
+        public Collection<Proposal> collectProposals(JsonNode document, IPath path) {
+            final Collection<Proposal> results = Lists.newArrayList();
+            if (value() == null) {
+                return results;
+            }
+
+            final List<JsonNode> parameters = document.findValues(value());
```

Comment:
`findValues` is a 
> Method for finding JSON Object fields with specified name, and returning found ones as a List.

As a result, we are getting all values of `operationId`, even those which are not in an operation. It can result in duplicate values or in invalid values.
## Invalid values:
<img width="291" alt="screen shot 2017-05-24 at 2 53 36 pm" src="https://cloud.githubusercontent.com/assets/644582/26420528/d3c15b1e-4090-11e7-9d61-83cfc0f8fd8a.png">

Model:
```yaml
openapi: "3.0.0"
info:
  title: Link operationID
  version: "1.0.0"  
  
paths: {}

components:
  links:
    UserRepositories:
      # returns array of '#/components/schemas/repository'
      operationId: "here"
      parameters:
        username: $response.body#/username
        
  schemas: 
    operationId: 
      type: object
      properties: 
        username: 
          type: string
        uuid: 
          type: string
```

## Duplicate values
<img width="514" alt="screen shot 2017-05-24 at 2 33 37 pm" src="https://cloud.githubusercontent.com/assets/644582/26420552/e3577de2-4090-11e7-804d-7231d8317b32.png">

Model:
```yaml
openapi: "3.0.0"
info:
  title: Link operationID
  version: "1.0.0"  
  
paths: 

  /2.0/users/{username}: 
    get: 
      operationId: getUserByName
      parameters: 
      - name: username
        in: path
        required: true
        schema:
          type: string
      responses: 
        '200':
          description: The User
          content:
            application/json:
              schema: 
                $ref: '#/components/schemas/user'
          links:
            userRepositories:
              $ref: '#/components/links/UserRepositories'
              
  /2.0/repositories/{username}:
    get:
      operationId: getRepositoriesByOwner
      parameters:
        - name: username
          in: path
          required: true
          schema:
            type: string
      responses:
        '200':
          description: repositories owned by the supplied user
          content: 
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/repository'
          links:
            userRepository:
              $ref: '#/components/links/UserRepository'
              
  /2.0/repositories/{username}/{slug}: 
    get: 
      operationId: getRepository
      parameters:  []
#        - name: username
#          type: string
#          in: path
#          required: true
#          schema:
#            type: string
#        - name: slug
#          in: path
#          required: true
#          schema:
#            type: string
      responses: 
        '200':
          description: The repository
          content:
            application/json: 
              schema: 
                $ref: '#/components/schemas/repository'
          links:
            repositoryPullRequests:
              $ref: '#/components/links/RepositoryPullRequests'
  /2.0/repositories/{username}/{slug}/pullrequests: 
    get: 
      operationId: getPullRequestsByRepository
      parameters: 
      - name: username
        in: path
        required: true
        schema:
          type: string
      - name: slug
        in: path
        required: true
        schema:
          type: string
      - name: state
        in: query
        schema:
          type: string
          enum: 
            - open
            - merged
            - declined
      responses: 
        '200':
          description: an array of pull request objects
          content:
            application/json: 
              schema: 
                type: array
                items: 
                  $ref: '#/components/schemas/pullrequest'
  /2.0/repositories/{username}/{slug}/pullrequests/{pid}: 
    get: 
      operationId: getPullRequestsById
      parameters: 
      - name: username
        in: path
        required: true
        schema:
          type: string
      - name: slug
        in: path
        required: true
        schema:
          type: string
      - name: pid
        in: path
        required: true
        schema:
          type: string
      responses: 
        '200':
          description: a pull request object
          content:
            application/json: 
              schema: 
                $ref: '#/components/schemas/pullrequest'
          links:
            pullRequestMerge:
              $ref: '#/components/links/PullRequestMerge'
  /2.0/repositories/{username}/{slug}/pullrequests/{pid}/merge: 
    post: 
      operationId: mergePullRequest
      parameters: 
      - name: username
        in: path
        required: true
        schema:
          type: string
      - name: slug
        in: path
        required: true
        schema:
          type: string
      - name: pid
        in: path
        required: true
        schema:
          type: string
      responses: 
        '204':
          description: the PR was successfully merged
          
components:
  links:
    UserRepositories:
      # returns array of '#/components/schemas/repository'
      operationId: getRepositoriesByOwner
      parameters:
        username: $response.body#/username
        
    UserRepository:
      # returns '#/components/schemas/repository'
      operationId: getRepository
      parameters:
        username: $response.body#/owner/username
        slug: $response.body#/slug
        
    RepositoryPullRequests:
      # returns '#/components/schemas/pullrequest'
      operationId: getPullRequestsByRepository
      # JSON Schema does not work here
#        parameters: 
#           username: $response.body#/owner/username
#           slug: $ref: $response.body#/slug
          
    PullRequestMerge:
      # executes /2.0/repositories/{username}/{slug}/pullrequests/{pid}/merge
      operationId: mergePullRequest
      parameters:
        username: $response.body#/author/username
        slug: $response.body#/repository/slug
        pid: $response.body#/id
        
  schemas: 
    user: 
      type: object
      properties: 
        username: 
          type: string
        uuid: 
          type: string
    repository: 
      type: object
      properties: 
        slug: 
          type: string
        owner: 
          $ref: '#/components/schemas/user'
    pullrequest: 
      type: object
      properties: 
        id: 
          type: integer
        title: 
          type: string
        repository: 
          $ref: '#/components/schemas/repository'
        author: 
          $ref: '#/components/schemas/user'
```